src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -202,6 +211,33 @@ public class RedundantModifierCheck
         TokenTypes.ABSTRACT,
     };
 
+    /**
+     * Constant for jdk 21 version number.
+     */
+    private static final int JDK_21 = 21;
+
+    /**
+     * Set the JDK version that you are using.
+     *
+     */
+    private String jdkVersion = String.valueOf(JDK_21);
```

Comment:
```suggestion
    /**
     * Set the JDK version that you are using.
     *
     */
    private String jdkVersion = String.valueOf(LEVEL_21.intValue());
```

Reply:
This will probably make things a bit easier to reason about and help us to remain "stateless"

Reply:
Ok but still tests require this check to be filesateful. It is due to the new javaLanguageLevel field

Reply:
ah shoot, right